packages/xod-project/src/utils.js

```diff
@@ -76,6 +79,36 @@ export const defaultValueOfType = def(
   R.flip(R.prop)(CONST.DEFAULT_VALUE_OF_TYPE)
 );
 
+export const getTypeFromLiteral = def(
+  'getTypeFromLiteral :: DataValue -> Either Error DataType',
+  literal => {
+    if (!R.is(String, literal))
+      return fail('BAD_LITERAL_VALUE', { value: literal });
+
+    if (isAmong(['True', 'False'], literal))
+      return Either.of(CONST.PIN_TYPE.BOOLEAN);
+
+    if (isAmong(R.values(CONST.INPUT_PULSE_PIN_BINDING_OPTIONS), literal))
+      return Either.of(CONST.PIN_TYPE.PULSE);
+
+    if (R.test(/^".*"$/gi, literal)) return Either.of(CONST.PIN_TYPE.STRING);
+
+    if (!isNaN(parseFloat(literal))) return Either.of(CONST.PIN_TYPE.NUMBER);
```

Comment:
`NaN` which is NaN is a valid literal value. Could you reuse what @brusherru did for validating number literals?

Reply:
Fixed!